### ISSUE TITLE
ObjectListing in version 1.10.45 does not URL decode prefix

### ISSUE THREAD

**Issue by alexkuang** [aws/aws-sdk-java#611]
In version 1.10.45, `prefix` is not URL-decoded.  This causes `s3.listNextBatchOfObjects()` to break, since `nextMarker` is decoded.  Example:

``` java
// existing s3 objects:
// s3://my-bucket/reports/year=2015/report.tsv
// s3://my-bucket/reports/year=2015/report2.tsv

ObjectListing o = s3.listObjects(req); // request with bucket="mybucket", prefix="/reports/year=2015", limit=1
System.out.println(o.getPrefix());
System.out.println(o.getNextMarker());

// output in 1.10.32
// reports/year=2015
// reports/year=2015/report.tsv

// output in 1.10.45
// reports/year%3D2015
// reports/year=2015/report.tsv

// returns nothing in 1.10.45, likely because of encoded prefix + decoded next marker
s3.listNextBatchOfObjects(o);
```

**Comment by manikandanrs** [comment]
This is a bug that has been fixed in 1.10.48

https://aws.amazon.com/releasenotes/Java/9020083571143869

**Comment by manikandanrs** [comment]
Please upgrade to the latest version of SDK and let us know if the issue still exists.
